sympy/assumptions/handlers/matrices.py

```diff
@@ -38,6 +38,8 @@ def MatMul(expr, assumptions):
         factor, mmul = expr.as_coeff_mmul()
         if all(ask(Q.symmetric(arg), assumptions) for arg in mmul.args):
             return True
+        if ask(Q.diagonal(expr), assumptions):
```

Comment:
When we ask `ask(Q.symmetric(V1.T*(V1 + V2)))` it never checks `Q.diagonal(V1.T*(V1 + V2))`, hence returns `None`.

However, inference works well by providing assumption explicitly `ask(Q.symmetric(M), Q.diagonal(M))`